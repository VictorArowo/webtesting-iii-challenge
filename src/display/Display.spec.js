import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Display from './Display';

let wrapper;

beforeEach(() => {
  wrapper = rtl.render(<Display closed={true} locked={true} />);
});

afterEach(rtl.cleanup);

describe('Display component', () => {
  it('matches the snapshot!', () => {
    expect(wrapper.container).toMatchSnapshot();
  });

  it('displays closed if closed prop is true', () => {
    expect(wrapper.queryByText(/closed/i)).toBeInTheDocument();
  });

  it('displays locked if locked prop is true', () => {
    expect(wrapper.queryByText(/\blocked\b/i)).toBeInTheDocument();
  });
  it('displays open if closed prop is false', () => {
    wrapper = rtl.render(<Display closed={false} locked={false} />);
    expect(wrapper.queryByText(/open/i)).toBeInTheDocument();
  });

  it('displays unlocked if locked prop is false', () => {
    wrapper = rtl.render(<Display closed={false} locked={false} />);
    expect(wrapper.queryByText(/unlocked/i)).toBeInTheDocument();
  });
});
