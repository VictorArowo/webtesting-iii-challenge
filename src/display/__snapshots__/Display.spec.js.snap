// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component matches the snapshot! 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
</div>
`;

exports[`Display component matches the snapshot! 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led red-led"
    >
      Locked
    </div>
    <div
      class="led red-led"
    >
      Closed
    </div>
  </div>
</div>
`;
